packages/components/src/DataList/components/DataListAction/DataListAction.tsx

```diff
@@ -24,10 +24,13 @@ export function DataListAction<T extends DataListObject>({
 
   const color = destructive ? "critical" : "heading";
 
+  const actionLabel =
+    typeof label === "function" && activeItem ? label(activeItem) : label;
+
   return (
     <button type="button" className={styles.action} onClick={handleClick}>
       <Typography textColor={color}>
-        <span className={styles.label}>{label}</span>
+        <span className={styles.label}>{actionLabel as string}</span>
```

Comment:
not super happy with this, but TS was not understanding that `actionLabel` was a string when `typeof label === "function"` :/